### PR TITLE
Login

### PR DESCRIPTION
**Pull request by jakeheft**
- Ability to login with email and password
- Will confirm validity of email and password to database
-  Redirected to profile if already logged in

### CODE REVIEW

**Review by GarrettCottrell — APPROVED** [review]
Looks good - worked on this driver/nav with Jake.

File: config/routes.rb [hunk 1]
```diff
@@ -34,7 +34,19 @@
...
 
+  namespace :merchant do
+    get "/", to: "dashboard#index"
+  end
+
+  namespace :admin do
+    get '/', to: "dashboard#index"
+  end
+
   get "/register", to: "users#new"
```
Comment (GregJMitchell) [inline config/routes.rb:45]: Nice use of namespacing!

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 
-  helper_method :cart
+  helper_method :cart, :current_user
 
   def cart
     @cart ||= Cart.new(session[:cart] ||= Hash.new(0))
   end
 
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
```
Comment (abcdefghijohn) [inline app/controllers/application_controller.rb:11]: Like the use of memoization and guard statement here.

**Review by GregJMitchell — APPROVED** [review]
Looks Great!

**Review by abcdefghijohn — APPROVED** [review]
Looks great, lets work on merge conflicts together!

**Pull request merged**